brave/brave-core
# Using synchronous preference deletion for wallet reset

Fixes brave/brave-browser#8454
Fixes brave/brave-browser#8747

This also correctly deletes the preferences on Windows builds, so by consequence brave/brave-browser#8454 was fixed as well. `brave_wallet_delegate` is no longer needed.


Resolves 

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:
1. On a fresh profile, enable crypto wallets
2. Create a wallet. 
3. Call `chrome.braveWallet.resetWallet()`
4. Confirm that the extension is completely reset. You will have to reload the tab, but that action will now be handled client side.

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
